bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/AuditorElector.java

```diff
@@ -238,8 +241,16 @@ public void shutdown() throws InterruptedException {
                 return;
             }
             // close auditor manager
-            submitShutdownTask();
-            executor.shutdown();
+            try {
+                submitShutdownTask().get(10, TimeUnit.SECONDS);
+                executor.shutdown();
+            } catch (ExecutionException e) {
+                LOG.warn("Failed to close auditor manager", e);
+                executor.shutdownNow();
+            } catch (TimeoutException e) {
+                LOG.warn("Failed to close auditor manager in 10 seconds", e);
+                executor.shutdownNow();
```

Comment:
I wonder if it would be necessary to set running to false in these exceptional cases so that the other thread that checks that value would exit?